src/main/java/com/hazelcast/remotecontroller/HazelcastCloudManager.java

```diff
@@ -0,0 +1,338 @@
+package com.hazelcast.remotecontroller;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.*;
+import okhttp3.Response;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import net.lingala.zip4j.ZipFile;
+import org.python.jline.internal.Log;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.*;
+import java.util.concurrent.TimeUnit;
+
+public class HazelcastCloudManager {
+    private static final OkHttpClient client = new OkHttpClient();
+    private String baseUrl;
+    private URI uri;
+    private final ObjectMapper mapper = new ObjectMapper();
+    private String bearerToken;
+    private static final Logger LOG = LogManager.getLogger(Main.class);
+    private static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+    private Call call;
+
+    public HazelcastCloudManager() {
+        if(System.getenv("BASE_URL") != null && System.getenv("API_KEY") != null && System.getenv("API_SECRET") != null)
+            loginToHazelcastCloud(System.getenv("BASE_URL"), System.getenv("API_KEY"), System.getenv("API_SECRET"));
+        else
+            LOG.info("Hazelcast cloud credentials are not set as env variable, please call login method first or cloud API cannot be used with RC");
+    }
+
+    public void loginToHazelcastCloud(String baseUrl, String apiKey, String apiSecret) {
+        this.uri = URI.create(baseUrl + "/api/v1");
+        this.baseUrl = baseUrl;
+        bearerToken = getBearerToken(apiKey, apiSecret);
+        if(bearerToken == null)
+            LOG.error("Login failed");
+    }
+
+    public CloudCluster createHazelcastCloudStandardCluster(String hazelcastVersion, boolean isTlsEnabled) {
+        try {
+            String clusterName = "test-cluster-" + System.currentTimeMillis();
```

Comment:
It seems it is not posbbile to have such a case because when you send request rc to create a cluster, it waits until it is running state. For example when you send two requests to rc paralelly from nodejs, the second request will wait for the first one because HazelcastCloudManager instance is already creating the cluster for the first request. 

I think if we run more than 1 rc and send cluster creation at the same time, there cloud be very very small posibility to have a same name. But I agree it could be more usefull to use UUID, let me change it.

>Or can't the cloud create a cluster without requiring us to specify a cluster name(with a random name)?

Normally when you create cluster on UI it suggest a name but from API it requeries the name.